app/javascript/src/ActiveDocs/OAS3Autocomplete.ts

```diff
@@ -120,3 +120,19 @@ export const autocompleteOAS3 = async (response: SwaggerUIResponse, accountDataU
     text: JSON.stringify(body)
   }
 }
+
+/**
+ * Intercept and process the response made by Swagger UI
+ * Apply transformations (inject servers list and autocomplete data) to the response for OpenAPI spec requests, and
+ * keep the responses to the actual API calls (made through 'Try it out') untouched
+ * @param response response to the request made through Swagger UI
+ * @param specUrl URL of the OpenAPI specification
+ * @param accountDataUrl URL of the data for autocompletion
+ * @param serviceEndpoint Public Base URL of the gateway, that will replace the  URL in the "servers" object
+ */
+export const autocompleteInterceptor = (response: SwaggerUIResponse, accountDataUrl: string, serviceEndpoint: string, specUrl?: string): Promise<Response> | SwaggerUIResponse => {
+  if (specUrl !== response.url) {
+    return response
+  }
+  return autocompleteOAS3(response, accountDataUrl, serviceEndpoint)
+}
```

Comment:
if `specUrl === response.url` couldn't we return the response as well?

Reply:
If `specUrl === response.url` we need to first process the response to apply some changes to the content (modify "servers" field, add some auto-complete data), and as it needs to fetch some data via an HTTP call, the response needs to be a promise.